### PR TITLE
Remove "requestAdsEnabled" API

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/25128

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by tmancey** [comment]
@aseren could you please code review this change? It has been agreed with @jsecretan that we can remove this change for now.

File: browser/brave_ads/sources.gni [hunk 1]
```diff
@@ -73,8 +73,6 @@ if (is_android) {
     "//brave/browser/brave_ads/android/brave_ads_native_helper.h",
     "//brave/browser/brave_ads/background_helper/background_helper_android.cc",
     "//brave/browser/brave_ads/background_helper/background_helper_android.h",
-    "//brave/browser/brave_ads/brave_ads_host_android.cc",
```
Comment (aseren) [inline browser/brave_ads/sources.gni:76]: It also needed to move `brave/browser/brave_ads/brave_ads_host.cc` to `brave_browser_brave_ads_sources` section

File: browser/brave_content_browser_client.cc [hunk 2]
```diff
@@ -187,6 +187,7 @@ using extensions::ChromeContentBrowserClientExtensionsPart;
 #endif
 
 #if !BUILDFLAG(IS_ANDROID)
+#include "brave/browser/brave_ads/brave_ads_host.h"
```
Comment (aseren) [inline browser/brave_content_browser_client.cc:190]: we need it for `android` also

File: components/brave_ads/renderer/DEPS [hunk 3]
```diff
@@ -2,9 +2,7 @@ include_rules = [
   "+brave/components/brave_ads/common",
   "+brave/components/brave_search/common",
   "+content/public/renderer",
-  "+gin",
   "+mojo/public/cpp/bindings",
   "+services/network/public/cpp",
-  "+third_party/blink/public",
-  "+v8/include",
+  "+third_party/blink/public"
```
Comment (aseren) [inline components/brave_ads/renderer/DEPS:7]: we can leave comma at the end

File: browser/brave_ads/sources.gni (hunk 1 above)
Comment (zenparsing) [reply browser/brave_ads/sources.gni:76]: Thanks for catching this!

File: browser/brave_content_browser_client.cc (hunk 2 above)
Comment (zenparsing) [reply browser/brave_content_browser_client.cc]: Done.

File: components/brave_ads/renderer/DEPS (hunk 3 above)
Comment (zenparsing) [reply components/brave_ads/renderer/DEPS]: Fixed.

**Review by aseren — APPROVED** [review]

**Review by bridiver — APPROVED** [review]

**Review by deeppandya — APPROVED** [review]
Android changes LGTM

**Pull request merged**